src/ghga_transpiler/config.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Module to process config file"""
+
+from collections import Counter
+from collections.abc import Callable
+
+from pydantic import BaseModel, ConfigDict, Field, computed_field, model_validator
+
+from .exceptions import DuplicatedName
+from .transformations import to_attributes, to_list, to_snake_case, to_snake_case_list
+
+
+class ColumnProperties(BaseModel):
+    """A data model for column properties"""
+
+    model_config = ConfigDict(populate_by_name=True)
+
+    sheet_name: str = Field(..., alias="sheet")
+    column_name: str = Field(..., alias="column")
+    multivalued: bool
+    type: str
+    ref_class: str | None
+    ref_id: str | None = Field(..., alias="ref_class_id_property")
+    enum: bool
+    required: bool
+
+    @computed_field  # type: ignore [misc]
```

Comment:
Do you intend to serialize `ColumnProperties` data?

Reply:
No, but as far as I understand `@computed_field` provides an advantage over using only `@property` since the value of the `transformation` attribute will be calculated and assigned as soon as an object of the class is initialized. In a way, this is used to replace how we used to utilize the `@model_validator(mode="after")` to set a field based on the other fields in pydantic v1.